components/uniset/src/uniset.rs

```diff
@@ -89,6 +90,7 @@ impl UnicodeSet {
     pub fn from_inversion_list(inv_list: Vec<u32>) -> Result<Self, UnicodeSetError> {
         if is_valid(&inv_list) {
             let size: usize = inv_list.chunks(2).map(|end_points| end_points[1] - end_points[0]).sum::<u32>() as usize;
+            // TODO: how to convert a `Vec<u32>` into ZeroVec?
```

Comment:
This function `from_inversion_list` takes an inversion list of `Vec<u32>` and needs to turn that into a ZeroVec inversion list when constructing the UnicodeSet.

But my understanding is that ZeroVec is designed for taking a byte array and converting it via allocation to an aligned array/vector of other types/widths only at the time it is needed. When the input is a non-byte slice, then an allocation is required to convert it into the backing byte array.

So is there a way to convert the `Vec<u32>` into the `&[u8]` without allocations?

Reply:
You should change the function to take `ZeroVec` instead of `Vec`.

If desired, you could keep these as two separate functions.